cli/azd/cmd/show.go

```diff
@@ -0,0 +1,188 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/internal"
+	"github.com/azure/azure-dev/cli/azd/pkg/commands"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+	"github.com/azure/azure-dev/cli/azd/pkg/infra"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+	"github.com/spf13/cobra"
+)
+
+func showCmd(rootOptions *internal.GlobalCommandOptions) *cobra.Command {
+	action := func(ctx context.Context, cmd *cobra.Command, args []string, azdCtx *azdcontext.AzdContext) error {
+		console := input.GetConsole(ctx)
+
+		formatter := output.GetFormatter(ctx)
+		writer := output.GetWriter(ctx)
+
+		// Right now this command is hidden and we only expect it to be called by tooling,
+		// which passes `--output json`. If for some reason someone ran it directly, just
+		// don't do anything.
+		if formatter.Kind() != output.JsonFormat {
+			return nil
+		}
+
+		if err := ensureProject(azdCtx.ProjectPath()); err != nil {
+			return err
+		}
+
+		env, ctx, err := loadOrInitEnvironment(ctx, &rootOptions.EnvironmentName, azdCtx, console)
+		if err != nil {
+			return fmt.Errorf("loading environment: %w", err)
+		}
+
+		prj, err := project.LoadProjectConfig(azdCtx.ProjectPath(), &env)
+		if err != nil {
+			return fmt.Errorf("loading project: %w", err)
+		}
+
+		res := showResult{
+			Services: make(map[string]showService, len(prj.Services)),
+		}
+
+		for name, svc := range prj.Services {
+			path, err := getFullPathToProjectForService(svc)
+			if err != nil {
+				return err
+			}
+
+			showSvc := showService{
+				Project: showServiceProject{
+					Path: path,
```

Comment:
I'm a little hesitant to call this "project path" given it means something different with the translation we're doing compared to `azure.yaml`. Are there more well-defined specifications on what this field should be? Is it supposed to always be "an absolute project file path"?

Reply:
I think I understand that this is a "validated" project path that can either be directory or file but is guaranteed to work. The translation we're doing was to avoid a single directory containing multiple project files -- I wonder if we should be doing validation and no translation. And if we do translation, we should have a consistent pattern to applying it.